### PR TITLE
Changed configuration

### PR DESCRIPTION
**Pull request by mohitsoni**
Added following config properties:
1. reservation ```boolean```
2. software binary ```string```
3. More required env vars

### CODE REVIEW

File: repo/packages/C/cassandra/0/marathon.json.mustache [hunk 1]
```diff
@@ -8,8 +8,8 @@
     0
   ],
   "uris": [
-    "{{resource.assets.uris.cassandra-mesos-0-2-0-1-tar-gz}}",
-    "{{resource.assets.uris.jre-7u76-linux-x64-tar-gz}}"
+    "{{cassandra.binary}}",
```
Comment (jsancio) [inline repo/packages/C/cassandra/0/marathon.json.mustache]: Why are we not grabbing it from the `resource.json`? This looks like it is going to break our packaging solution for hosting on premise and upgrade story.

**Comment by jsancio** [comment]
@mohitsoni Please create a brand new revision/release under `.../cassandra/1/` directory. We require that package revision/release are immutable.

File: repo/packages/C/cassandra/0/config.json [hunk 2]
```diff
@@ -91,6 +97,11 @@
           "type": "string",
           "default": "dcos"
         },
+        "binary": {
+          "description":"Cassandra binary",
+          "type":"string",
+          "default":"http://downloads.mesosphere.com/cassandra-mesos/artifacts/0.2.0-1/cassandra-mesos-0.2.0-1.tar.gz"
+        },
```
Comment (jsancio) [inline repo/packages/C/cassandra/0/config.json]: I don't recommend this. I this going to break our packaging for local universe (datacenters without access to that URL) and service upgrade.

File: repo/packages/C/cassandra/0/resource.json [hunk 3]
```diff
@@ -1,9 +1,9 @@
 {
   "assets": {
     "uris": {
-      "cassandra-mesos-0-2-0-1-tar-gz":
+      "cassandra-mesos-binary-tar-gz":
       "https://downloads.mesosphere.com/cassandra-mesos/artifacts/0.2.0-1/cassandra-mesos-0.2.0-1.tar.gz",
-      "jre-7u76-linux-x64-tar-gz":    "https://downloads.mesosphere.com/java/jre-7u76-linux-x64.tar.gz"
+      "jre-tar-gz":    "https://downloads.mesosphere.com/java/jre-7u76-linux-x64.tar.gz"
```
Comment (jsancio) [inline repo/packages/C/cassandra/0/resource.json]: If possible please don't modify the keys, we are going to build tools that can generate this file automatically from a directory. I think this change will make it hard to build such tools. This requires that the keys are generated from the filename.

File: repo/packages/C/cassandra/0/config.json (hunk 2 above)
Comment (mohitsoni) [inline repo/packages/C/cassandra/0/config.json]: Thanks for the feedback @jsancio. The intention behind doing this is basically allow integration tests over-ride the binary location. Doing so sets us free from maintaining a separate testing branch.

Is there any way I can override resource.json properties via options.json or some other mechanism ?

**Comment by mohitsoni** [comment]
Closing the PR post discussion with @jsancio. We'll explore an alternative approach to achieve our goals.

**Pull request closed without merging**